Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM jekyll/jekyll as build-env
+
+WORKDIR /src
+
+COPY Gemfile* ./
+RUN chmod 777 /src
+RUN  bundle install
+
+COPY . .
+RUN bundle exec jekyll build --baseurl ''
+
+FROM nginx:alpine
```

Comment:
Could you walk me through the use of Nginx here. Thanks!

Reply:
nginx is just a web server, to serve the static website generated by jekyll.  
do you have a specific question?

Reply:
@ammarnajjar yes, I was wondering why use nginx when jekyll can serve the build itself?

Reply:
you could do that, but it is not recommended.  
It is always recommended to build your web app (whatever that is) using a build image, then move the build outcome to a web server image and serve it from there for many reasons.  
The main two for me are:
- to decrease the produced image size: the image serving your web app does not need to have all the build tools in it, that's why I used `jekyll` image to build and a small `nginx` to serve.
- a web server is light weight and optimized to do just that!